src/flux/student-dashboard.coffee

```diff
@@ -40,6 +45,18 @@ StudentDashboardConfig = {
       longTitle = "#{data.course.name} | #{arrayToSentence(data.course.teacher_names)}"
       {longTitle, shortTitle}
 
+    canWorkTask: (event) ->
+      # only homework and readings can be worked on
+      _.contains(['homework', 'reading'], event.type) and
+        # readings can always be performed
+        (event.type is 'reading' or
+            # other types (homework) must be incomplete and not past due
+            (not event.complete and moment(event.due_at).isAfter(TimeStore.getNow())))
+
+    upcomingEvents: (courseId) ->
```

Comment:
This controls whether a task gets a onClick handler and navigates to it.  I'm not entirely sure of the logic behind it, but it seems close to what the task panel used.